nodebuilder/store.go

```diff
@@ -150,6 +151,16 @@ type fsStore struct {
 	dirLock *flock.Flock // protects directory
 }
 
+// IsOpened checks if a directory is in use by checking if a lock file exists.
+func IsOpened(path string) bool {
+	lockPath := lockPath(path)
+	if _, err := os.Stat(lockPath); errors.Is(err, os.ErrNotExist) {
+		return false
+	}
```

Comment:
This is not enough. We should check for actual locking besides the file using the flock library.

Reply:
yup working on that, just pushed initial

Reply:
added now